ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -278,29 +283,29 @@ showForSnippet x = printOutputable x
 
 mkModCompl :: T.Text -> CompletionItem
 mkModCompl label =
-  CompletionItem label (Just CiModule) Nothing Nothing
+  CompletionItem label Nothing (Just CompletionItemKind_Module) Nothing Nothing
```

Comment:
:vomiting_face: maybe we could make these use record syntax...

Reply:
Yeah actually, this would be a nice place to create an instance for Default, and here just add the specific fields we are using.

Reply:
Yeah maybe, although maybe `emptyCompletionItem :: Text -> CompletionItem` that at least sets the non-optional label field is better.

Mostly I just think these massive positional constructor applications are hard to understand or update.

Reply:
Yes absolutely, I think this area that has the highest chance of containing a bug with the new hls and lsp is the constructor applications. Fields were switched with the new generated version, and all it needs is 2 fields with the same type to switch to get a bunch of errors that could be in untouched code.